splice_machine/src/main/java/com/splicemachine/derby/impl/storage/CheckTableJob.java

```diff
@@ -100,17 +99,21 @@ public Void call() throws Exception {
 
         String table = Long.toString(heapConglomId);
         Collection<PartitionLoad> partitionLoadCollection = EngineDriver.driver().partitionLoadWatcher().tableLoad(table, true);
+
         boolean distributed = false;
-        for (PartitionLoad load: partitionLoadCollection) {
-            if (load.getMemStoreSize() > 1*MB || load.getStorefileSize() > 1*MB)
-                distributed = true;
+        if (request.useSpark == null) {
+            for (PartitionLoad load : partitionLoadCollection) {
+                if (load.getMemStoreSize() > 1 * MB || load.getStorefileSize() > 1 * MB)
+                    distributed = true;
+            }
+        } else {
+            distributed = request.useSpark;
         }
         DataSetProcessor dsp = null;
         if (distributed) {
             SpliceLogUtils.info(LOG, "Run check_table on spark");
             dsp = EngineDriver.driver().processorFactory().distributedProcessor();
-        }
-        else {
+        } else {
             SpliceLogUtils.info(LOG, "Run check_table on region server");
             dsp = EngineDriver.driver().processorFactory().localProcessor(null, null);
         }
```

Comment:
This doesn't make much sense to me, CheckTableJob is already running in the OlapServer, as if it was going to be running in Spark. But here we are choosing whether to run it in Spark or the RS. If we choose RS then as far as I can see it's going to run on the OlapServer instead. That decision should have been made in SpliceAdmin.checkIndexesInDetail() when submitting the checkTableJob to the OlapServer

Reply:
This is not the problem I'm addressing in this PR. I'll file another Jira.